Community Toolbox/scripts/utils_CommunityToolboxDraw/utils_CommunityToolboxDraw.gml

```diff
@@ -0,0 +1,266 @@
+    if (_anglefrom == _angleto)
+        return;
+    
+    if (_outline && _angleto - _anglefrom >= 360) {
+        draw_arc(_x, _y, _radius, 0, 360, _precision);
+        return;
+    }
+    
+    // draw the actual pie
+    draw_primitive_begin(_outline ? pr_linestrip : pr_trianglefan);
+
+    draw_vertex(_x, _y);
+    draw_arc_vertices(_x, _y, _radius, _anglefrom, _angleto, _precision, false);
+    if (_outline)
+        draw_vertex(_x, _y);
+    
+    draw_primitive_end();
+}
+
+/// @func draw_chord(x,y,radius,anglefrom,angleto,[outline],[precision])
+/// @desc Draws a chord-like shape (i.e. a circle segment) around the given center, with the given radius and between given angles.
+/// @arg {Real} x               The x coordinate of the chord center.
+/// @arg {Real} y               The y coordinate of the chord center.
+/// @arg {Real} radius          The radius of the chord around the center.
+/// @arg {Real} anglefrom       The starting angle of the chord.
+/// @arg {Real} angleto         The ending angle of the chord.
+/// @arg {Bool} [outline]       Whether only the outline of the chord should be drawn or it should be filled.
+/// @arg {Real} [precision]     The precision of the circle the chord's arc is a part of.
+function draw_chord(_x, _y, _radius, _anglefrom, _angleto, _outline = false, _precision = 24) {
+    // making the initial "draw_arc" call to prepare its static struct
+    static draw_arc_init = draw_arc(0, 0, 0, 0, 0);
+    static normalize_angles = draw_arc.normalize_angles;
+    static draw_arc_vertices = draw_arc.draw_arc_vertices;
+    
+    // normalise the angles for easier processing
+    var _normangles = normalize_angles(_anglefrom, _angleto);
+    _anglefrom = _normangles[0];
+    _angleto = _normangles[1];
+    if (_anglefrom == _angleto)
+        return;
+    
+    // draw the actual circle segment
+    draw_primitive_begin(_outline ? pr_linestrip : pr_trianglefan);
+    draw_arc_vertices(_x, _y, _radius, _anglefrom, _angleto, _precision, _outline);
+    draw_primitive_end();
+}
+
+/// @func draw_arc_range(x,y,r1,r2,anglefrom,angleto,[outline],[precision])
+/// @desc Draws a band-like shape between two arcs around the given center, with the given radii and between given angles.
+/// @arg {Real} x               The x coordinate of the chord center.
+/// @arg {Real} y               The y coordinate of the chord center.
+/// @arg {Real} r1              The first radius to draw the band between.
+/// @arg {Real} r2              The second radius to draw the band between.
+/// @arg {Real} anglefrom       The starting angle of the arcs.
+/// @arg {Real} angleto         The ending angle of the arcs.
+/// @arg {Bool} [outline]       Whether only the outline of the band should be drawn or it should be filled.
+/// @arg {Real} [precision]     The precision of the circles the band is drawn between.
+function draw_arc_range(_x, _y, _r1, _r2, _anglefrom, _angleto, _outline = false, _precision = 24) {
```

Comment:
If you're still not fond of this name, you could consider `draw_washer`. There's [mathematical precedent](https://en.wikipedia.org/wiki/Disc_integration#Washer_method) for calling that sort of shape a "washer", and it would be very difficult to confuse such a name with anything else.

Reply:
At this rate, I'd rather make a separate `draw_ring` function (for drawing a full hollowed disk shape) and then name this one `draw_ring_arc`. ^^'
I'm fairly certain most people can imagine all kinds of objects to give "washer" name to with few actually matching rings in appearance. Though that also comes from a non-native speaker, who'd more readily thing of dishwasher rather than "washer" on its own. 😛

Reply:
That's fair. For people who _aren't_ grouchy old math teachers like me, I suppose "washer" does conjure images of a machine. Was just a passing thought; I don't know that there _is_ a good name for this, so it's probably fine as is.